3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py

```diff
@@ -63,46 +64,91 @@
 
 all_GSEA_results = []
 all_signatures = []
-#negative_control = []
-for col in signature.iloc[:,1:50].columns:
+neg_GSEA_results = []
+negative_control = []
+
+for col in signature.iloc[:,1:101].columns:
     df = signature.iloc[:,[0,int(col)]]
     result = blitz.gsea(df, library)
     all_GSEA_results.append(result.assign(z_dim=f"z_{col}"))
     all_signatures.append(df)
-    #add line here to create a negative control--this will be a random scramble of the gene scores
+    neg_df = df.copy()
+    neg_df[col] = neg_df[col].sample(frac=1).reset_index(drop=True)
+    neg_result = blitz.gsea(neg_df, library)
+    neg_GSEA_results.append(neg_result.assign(z_dim=f"z_{col}"))
+    negative_control.append(neg_df)
+
 all_GSEA_results
+neg_GSEA_results
 
 
 # In[7]:
 
 
 # stack up all of the results to be analyzed
 all_GSEA_results= pd.concat(all_GSEA_results)
+neg_GSEA_results = pd.concat(neg_GSEA_results)
 
 
 # In[8]:
 
 
 # sort by what you want to evaluate
-all_GSEA_results.sort_values(by='es', ascending=False)
+all_GSEA_results['rank'] = (-np.log10(all_GSEA_results.pval))*(all_GSEA_results.es)
```

Comment:
can you add documentation describing what the column `rank` is?

Also, checkout the function [`pandas.DataFrame.assign()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.assign.html) - I generally prefer it to creating a new column through subsetting.

Reply:
I was inspired by a video I watched detailing gsea plotting and was initially planning on including the `rank` column as a way to rank the enrichment of each pathway. However, I realized this was unnecessary because unlike the gsea package used in the video, the rank column is not a necessary input for the blitzgsea plot function.